### PR TITLE
feat: Add support for D_IIII

### PR DESCRIPTION
**Pull request by phako**
Simple support that maps a D_IIII to an array

### CODE REVIEW

**Comment by phako** [comment]
Needs https://github.com/ChimeraTK/ControlSystemAdapter/pull/103

**Review by mhier — APPROVED** [review]

File: src/DoocsPVFactory.cc [hunk 1]
```diff
@@ -365,6 +366,43 @@ namespace ChimeraTK {
...
+    doocsPV->setMatchingMode(iiiiDescription.dataMatching);
+
+    // set macro pulse number source, if configured
+    if(iiiiDescription.macroPulseNumberSource.size() > 0) {
+      auto mpnSource = _controlSystemPVManager->getProcessVariable(iiiiDescription.macroPulseNumberSource);
+      auto mpnDecorated = getTypeChangingDecorator<int64_t>(mpnSource, DecoratorType::C_style_conversion);
+      if(mpnDecorated->getNumberOfSamples() != 1) {
+        throw ChimeraTK::logic_error("The property '" + mpnDecorated->getName() +
+            "' is used as a macro pulse number source, but it has an array "
+            "length of " +
```
[the remark this answers is not in the document]
Comment (phako) [reply src/DoocsPVFactory.cc:386]: I tried format-ccode but it does not change this - and it is identical to the other instances of that error.

File: include/PropertyDescription.h [hunk 2]
```diff
@@ -254,6 +254,24 @@ namespace ChimeraTK {
...
   /********************************************************************************************************************/
 
+  struct IiiiDescription : public PropertyDescription, public PropertyAttributes {
+    ChimeraTK::RegisterPath iiiiSource;
+
+    IiiiDescription(ChimeraTK::RegisterPath const& iiiiSource_ = "", std::string const& location_ = "",
+        std::string const& name_ = "")
+    : PropertyDescription(location_, name_), iiiiSource(iiiiSource_){};
+
+    const std::type_info& type() const override { return typeid(IiiiDescription); }
```
[the remark this answers is not in the document]
Comment (phako) [reply include/PropertyDescription.h]: I have removed type()

**Comment by phako** [comment]
Ah there was no +, fixed it

**Pull request merged**